slave/process-mailaliases-generic/bin/process-mailaliases_generic.sh

```diff
@@ -19,28 +19,44 @@ ETC_ALIASESD_DIR="/etc/aliases.d/"
 PERMISSIONS="644"
 
 function process {
+
+	### prepare destination file variables
+	if [ -z "$DST_FILE" ]; then
+		DST_FILE="${ETC_ALIASESD_DIR}/${GENERIC_FILE}"
+	fi
+	DST_FILE_DIR=$(dirname "$DST_FILE")
+	DST_FILE_NAME=$(basename "$DST_FILE")
+
 	### Create lock
 	create_lock
 
 	### Set permisson for special case if destination file not exists
 	catch_error E_PERMISSIONS chmod "${PERMISSIONS}" "${FROM_PERUN}"
 
-	### try if /etc/aliases.d/ exists, if not, end with error
-	if [ ! -d "${ETC_ALIASESD_DIR}" ]; then
-		log_msg E_NOTEXISTING_ALIASESD_DIR
+	### try if destination directory exists, if not, end with error
+	if [ ! -d "${DST_FILE_DIR}" ]; then
+		log_msg E_NOTEXISTING_DST_FILE_DIR
 	fi
 
 	### try if file /etc/aliases exists, if not, end with error
 	if [ ! -f "${ETC_ALIASES}" ]; then
 		log_msg E_NOTEXISTING_ALIASES_FILE
 	fi
 	
-	### Looking for duplicits in possible places (except perun_generic_file)
-	### take all files from /etc/aliases.d/ except old generic_file and all files with extension .db
-	fail_if_duplicits_found "${ETC_ALIASES}" "${FROM_PERUN}" `find "${ETC_ALIASESD_DIR}" -type f -not -name ${GENERIC_FILE} -not -name "*.db" -print`
+	### Looking for duplicits in possible places
+	### Checks /etc/aliases.d/ except old generic_file, all files with extension .db and destination file (if it is set)
+	### If there is a destination file set inside different directory than /etc/aliases.d/, check that directory as well
+	if [ "${DST_FILE}" = "${ETC_ALIASESD_DIR}/${GENERIC_FILE}" ]; then
+		fail_if_duplicits_found "${ETC_ALIASES}" "${FROM_PERUN}" `find "${ETC_ALIASESD_DIR}" -type f -not -name "${GENERIC_FILE}" -not -name "*.db" -print`
+	elif [ "${DST_FILE_DIR}" = "${ETC_ALIASESD_DIR}" ]; then
+		fail_if_duplicits_found "${ETC_ALIASES}" "${FROM_PERUN}" `find "${ETC_ALIASESD_DIR}" -type f -not -name "${DST_FILE_NAME}" -not -name "${GENERIC_FILE}" -not -name "*.db" -print`
+	else
+		fail_if_duplicits_found "${ETC_ALIASES}" "${FROM_PERUN}" `find "${DST_FILE_DIR}" -type f -not -name "${DST_FILE_NAME}" -not -name "*.db" -print`
```

Comment:
I'm not sure it works as intended. We should probably check all files at once and not only each folder, since duplicate entries can be between folders too.